### PR TITLE
[#11,irods/irods#7263] CMake consistency sweep

### PR DESCRIPTION
**Pull request by SwooshyCueb**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.12.0 FATAL_ERROR)
 
-option(IRODS_ENABLE_430_COMPATIBILITY "Enables compatibility with iRODS 4.3.0." YES) # Default to YES for community.
+option(IRODS_ENABLE_430_COMPATIBILITY "Enables compatibility with iRODS 4.3.0." NO) # Default to YES for community.
```
Comment (korydraughn) [inline CMakeLists.txt:3]: Tell me about this?

File: rule_engine/CMakeLists.txt [hunk 2]
```diff
@@ -5,16 +5,6 @@ add_library(
...
 
-target_compile_options(
-  ${IRODS_RULE_ENGINE_PLUGIN_NAME}
-  PRIVATE
-  -Wall
-  -Wextra
-  #-Werror
-  -Wconversion
-  -Wshadow
-  -Wpedantic)
```
Comment (korydraughn) [inline rule_engine/CMakeLists.txt:16]: Are these set by something else?

File: CMakeLists.txt (hunk 1 above)
Comment (SwooshyCueb) [reply CMakeLists.txt:3]: This wasn't supposed to make it into the PR, will fix

File: rule_engine/CMakeLists.txt (hunk 2 above)
Comment (SwooshyCueb) [reply rule_engine/CMakeLists.txt:16]: Yes, they are already set here:
https://github.com/irods/irods_api_plugin_genquery2/blob/b314f1eec2b32a3a6b59f52fefe29a98d86d2aba/CMakeLists.txt#L62

Comment (korydraughn) [reply rule_engine/CMakeLists.txt:16]: Cool. Thanks.

**Review by korydraughn — APPROVED** [review]
Add the pounds when ready.

**Comment by SwooshyCueb** [comment]
\#'d

**Pull request merged**